extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,20 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
-
+      const self = this;
```

Comment:
I am sorry I don't understand. What does `this` refer to once the context changes? Line 226 uses `this` while line 228 uses `self`. Are the two calls being called on two different objects?